web/src/test/java/com/graphhopper/http/resources/RouteResourceCustomModelTest.java

```diff
@@ -242,34 +230,71 @@ public void testSubnetworkRemovalPerProfile() {
                 " \"profile\": \"car_no_unclassified\"," +
                 " \"ch.disable\": true" +
                 "}";
-        JsonNode jsonNode = query(body, 200).readEntity(JsonNode.class);
-        JsonNode path = jsonNode.get("paths").get(0);
+        JsonNode path = getPath(body);
         assertEquals(8754, path.get("distance").asDouble(), 5);
 
         // CH
         body = "{\"points\": [[11.556416,50.007739], [11.528864,50.021638]]," +
                 " \"profile\": \"car_no_unclassified\"" +
                 "}";
-        jsonNode = query(body, 200).readEntity(JsonNode.class);
-        path = jsonNode.get("paths").get(0);
+        path = getPath(body);
         assertEquals(8754, path.get("distance").asDouble(), 5);
 
         // different profile
         body = "{\"points\": [[11.494446, 50.027814], [11.511483, 49.987628]]," +
                 " \"profile\": \"custom_bike2\"," +
                 " \"ch.disable\": true" +
                 "}";
-        jsonNode = query(body, 200).readEntity(JsonNode.class);
-        path = jsonNode.get("paths").get(0);
+        path = getPath(body);
         assertEquals(5370, path.get("distance").asDouble(), 5);
     }
 
+    @Test
+    public void customBikeWithHighSpeed() {
+        // encoder.getMaxSpeed is 30km/h and limit_to statement is 40km/h in server-side custom model
+        String jsonQuery = "{" +
+                " \"points\": [[11.58199, 50.0141], [11.5865, 50.0095]]," +
+                " \"profile\": \"custom_bike3\"," +
+                " \"custom_model\": { " +
+                "   \"speed\": [{" +
+                "       \"if\": \"true\", \"limit_to\": 55" +
+                "     }] " +
+                " }," +
+                " \"ch.disable\": true" +
+                "}";
+
+        final Response response = query(jsonQuery, 400);
+        JsonNode json = response.readEntity(JsonNode.class);
+        assertTrue(json.get("message").asText().startsWith("vehicle speed of CustomModel in query cannot be bigger than 40"), json.get("message").asText());
```

Comment:
With all the info I have now I would rather remove this error and silently ignore statements with a too big limit_to value, to easier maintain backward compatibility. Because whenever we increase the vehicle's max_speed we could break query-side custom models.

WDYT @easbar?

Reply:
To be honest I never saw so much value in this check. Your point about backward compatibility seems reason enough to remove it with my current understanding.

One issue that this check somewhat helps with is that the max speed is hardly visible for the user, but I am sure there are better ways to improve this than having this check.